### ISSUE TITLE
[TASK] Cleanup/Refactor invocation of compiler pipeline

### ISSUE THREAD

**Issue by keyseven123** [nebulastream/nebulastream#349]
### Description

Once #102 is merged into the main, we can refactor the invocation of the compiler query. Currently, there are a lot of unnecessary calls, i.e.,
![image](https://github.com/user-attachments/assets/f5b0f282-a349-4b01-a1c2-0f7e8aafd7b0)

It looks like they are leftovers from the legacy (before nautilus) query compiler.
